### ISSUE TITLE
Regex doesn't handle beginnings in backtracing

### ISSUE THREAD

**Issue by Pandapip1** [oven-sh/bun#5288]
### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 unknown

### What steps can reproduce the bug?

I was originally using NodeJS for my project. I have the following regex to extract github usernames from an author string:

```js
let reviewers = "Hello World (@hello), Neet (@neet), Honk, Foo <bar@example.com>".match(/(?<=(?:^|,)[^<(]+\(@)(.*?)(?=\)\s*(<(("[^"]+")|([\w.]+)@[\w.]+)>)?(?:$|,))/g) || [];
```

### What is the expected behavior?

In NodeJS:

```js
> console.dir(reviewers)
[ "hello", "neet" ]
```

### What do you see instead?

In Bun:

```js
> console.dir(reviewers)
[ "neet" ]
```

**Comment by Pandapip1** [comment]
Found the issue. It's a combination of back/forward tracing and the start/end symbols. The following code produces correct results until this is fixed:

```js
let reviewers = `,${frontMatter.attributes.author},`?.match(/(?<=(?:^|,)[^<(]+\(@)(.*?)(?=\)\s*(<(("[^"]+")|([\w.]+)@[\w.]+)>)?(?:$|,))/g) || [];
```

**Comment by Electroid** [comment]
This might be fixed by #5167. @dylan-conway want to check?

**Comment by Pandapip1** [comment]
> This might be fixed by https://github.com/oven-sh/bun/pull/5167. @dylan-conway want to check?

If it is, would it be possible to release a version 1.0.2 sooner rather than later?

**Comment by dylan-conway** [comment]
Unfortunately this is a bug in JSC and #5167 does not fix it.

**Comment by dylan-conway** [comment]
This has been fixed in the latest version of safari. We will have this bug fix when we upgrade our WebKit fork

**Comment by Pandapip1** [comment]
> We will have this bug fix when we upgrade our WebKit fork

When will that be?

**Comment by dylan-conway** [comment]
fixed in v1.0.3
